bpo-34271: Add ssl debugging helpers (GH-10031)

tiran opened the pull request:
The ssl module now can dump key material to a keylog file and trace TLS
protocol messages with a tracing callback. The default and stdlib
contexts also support SSLKEYLOGFILE env var.

- [ ] API review
- [ ] whatsnew entry
- [ ] More tests

<!-- issue-number: [bpo-34271](https://bugs.python.org/issue34271) -->
https://bugs.python.org/issue34271
<!-- /issue-number -->

tiran committed:
bpo-34271: Add ssl debugging helpers

The ssl module now can dump key material to a keylog file and trace TLS
protocol messages with a tracing callback. The default and stdlib
contexts also support SSLKEYLOGFILE env var.

reaperhulk commented on Doc/library/ssl.rst:
```diff
@@ -1901,6 +1931,20 @@ to speed up repeated connections from the same clients.
+   Write TLS keys to a keylog file, whenever key material is generated or
+   received. The keylog file is designed for debugging purposes only. The
+   file format is specified by NSS and used by many traffic analyzers such
+   as Wireshark. The log file is opened in append-only mode. Writes a
```
s/Writes a/writes are

tiran committed:
Make msg_callback a private debug function

The msg_callback and related enums are now private members. The feature
is designed for internal debugging and not for end users.

Signed-off-by: Christian Heimes <christian@python.org>
